ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/RecordFormatter.java

```diff
@@ -248,7 +251,7 @@ String format(final Bytes bytes) {
     ) {
       try {
         final Object result = deserializer.deserializer.deserialize(topicName, bytes.get());
-        return Optional.of(result == null ? "<null>" : result.toString());
+        return Optional.of(result == null ? "<null>" : deserializer.format(result));
```

Comment:
This call to `format` seems completely superfluous.  This method now calls `deserialize` on the deserializer and then immediately passes the result back  to `format` on the same deserializer.  Why not just have `deserialize` return the formatted result?

In the case of the new protobuf format, all we need to do is pass a factory method to the constructor of the `PROTOBUF` enum that wraps the `KafkaProtobufDeserializer` instance in one that applies the format, in a similar way to `newJsonDeserializer`.

```java
   private static Deserializer<?> newProtobufDeserializer(final SchemaRegistryClient srClient) {

    final Printer printer = TextFormat.printer();
    final KafkaProtobufDeserializer<?> inner = new KafkaProtobufDeserializer<>(srClient);
    
    return (Deserializer<Object>) (topic, data) -> {
      final Message msg = inner.deserialize(topic, data);
      if (msg == null) {
        return null;
      }
      
      return printer.shortDebugString(msg);
    };
  }

...

 enum Format {
    AVRO(0, KafkaAvroDeserializer::new),
    PROTOBUF(0, RecordFormatter::newAvroDeserializer),
    JSON(RecordFormatter::newJsonDeserializer),
 ...
 } 
```

This avoids the need to complicate the code with another function call on _all_ deserializers...